cpp/daal/src/services/compiler/generic/env_detect_features.cpp

```diff
@@ -41,6 +46,28 @@
 void __daal_serv_CPUHasAVX512f_enable_it_mac();
 #endif
 
+#if defined(__ARM_ARCH)
```

Comment:
Split the file up into different arch implementations, and put these in appropriate sub-directories